### PR TITLE
Chang to merge

### PR DESCRIPTION
**Pull request by bendichter**
@luiztauffer I went through and handled all of the merge conflicts with the Chang Lab's master branch, but I wanted to run these changes by you before merging. When you have time, can you look at these and make sure that it doesn't break anything on your end?

### CODE REVIEW

**Comment by bendichter** [comment]
It looks like there were some issues merging these together. I'll look into these.

File: ecogvis/functions/misc_dialogs.py [hunk 1]
```diff
@@ -158,8 +172,11 @@ def __init__(self, parent, type):
         self.parent = parent
         self.type = type
         self.val = -1
-        self.text = QLabel("There is no Spectrum data for " + self.type + " data in the current NWB file.\n"
-                           "To calculate the Power Spectral Density, click Calculate.")
+        self.text = QLabel(
+            "There is no Spectrum data for " + self.type + " data in the "
+                                                           "current NWB "
+                                                           "file.\n" +
+            "To calculate the Power Spectral Density, click Calculate.")
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:179]: why this arrangement?

File: ecogvis/functions/misc_dialogs.py [hunk 2]
```diff
@@ -175,8 +192,10 @@ def __init__(self, parent, type):
         self.exec_()
 
     def run_estimation(self):
-        self.text.setText("Calculating the Power Spectral Density for " + self.type + " data.\n"
-                          "Please wait, this might take around 5 minutes.")
+        self.text.setText(
+            "Calculating the Power Spectral Density for " + self.type + " "
+                                                                        "data.\n" +
+            "Please wait, this might take around 5 minutes.")
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:198]: again, why it is disposed like that?

File: ecogvis/functions/misc_dialogs.py [hunk 3]
```diff
@@ -316,16 +341,19 @@ def __init__(self, parent):
 
         if 'ecephys' in parent.model.nwb.processing:
             # If there's no preprocessed data in NWB file
-            if 'LFP' not in parent.model.nwb.processing['ecephys'].data_interfaces:
+            if 'LFP' not in parent.model.nwb.processing[
+                'ecephys'].data_interfaces:
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:345]: some weird line breaks happening in many places

File: ecogvis/functions/misc_dialogs.py [hunk 4]
```diff
@@ -316,16 +341,19 @@ def __init__(self, parent):
...
                 self.disable_all()
                 text = "There's no preprocessed data in the current file.\n" \
                        "Run 'Preprocess' to generate it."
                 self.label_1.setText(text)
                 self.cancelButton.setEnabled(True)
             # If there's already Bandpower data in NWB file
-            if 'DecompositionSeries' in self.nwb.processing['ecephys'].data_interfaces:
+            if 'DecompositionSeries' in self.nwb.processing[
+                'ecephys'].data_interfaces:
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:353]: here again

File: ecogvis/functions/misc_dialogs.py [hunk 5]
```diff
@@ -341,14 +369,18 @@ def __init__(self, parent):
...
+               "file.\n" \
+               "It can be created from the bands shown in the table. " \
+               "The results will be saved in the current NWB file.\nDo you " \
+               "want to create it?"
         self.label_1.setText(text)
         self.runButton.setEnabled(True)
         # Populate table with values
-        self.tableWidget.setHorizontalHeaderLabels(['center [Hz]', 'sigma [Hz]'])
+        self.tableWidget.setHorizontalHeaderLabels(
+            ['center [Hz]', 'sigma [Hz]'])
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:383]: again

File: ecogvis/functions/misc_dialogs.py [hunk 6]
```diff
@@ -467,7 +500,8 @@ def __init__(self, parent):
     def check_hg_exists(self):
         if 'ecephys' in self.parent.model.nwb.processing:
             # If there's no preprocessed data in NWB file
-            if 'LFP' not in self.parent.model.nwb.processing['ecephys'].data_interfaces:
+            if 'LFP' not in self.parent.model.nwb.processing[
+                'ecephys'].data_interfaces:
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:504]: here again

File: ecogvis/functions/misc_dialogs.py [hunk 7]
```diff
@@ -1541,9 +1598,1119 @@ def paint(self):
...
+        left = p.getAxis('left')
+        left.setStyle(showValues=False)
+        left.setTicks([])
+        bottom = p.getAxis('bottom')
+        bottom.setStyle(showValues=True)
+        p.setLabel('bottom', 'Time', units='sec')
+
+
+# Creates Audio Event Detection window --------------------------------------
+class AudioEventDetection(QtGui.QDialog):
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:2280]: this has been moved out of here to its own module: 'audio_event_detection.py'

File: ecogvis/functions/misc_dialogs.py [hunk 8]
```diff
@@ -1541,9 +1598,1119 @@ def paint(self):
...
+                # Write file
+                io.write(nwb)
+            self.parent.model.refresh_file()
+            self.parent.model.SpeakerAndMicIntervalAdd()
+            self.parent.model.refreshScreen()
+        self.accept()
+
+
+# Runs 'detect_events' function, useful to wait for thread -----------------
+class EventDetectionFunction(QtCore.QThread):
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:2691]: this has been moved out of here as well

File: ecogvis/functions/misc_dialogs.py [hunk 9]
```diff
@@ -1541,9 +1598,1119 @@ def paint(self):
...
+        pg.ViewBox.__init__(self)
+        self.parent = parent
+        self.ch = ch
+
+    def mouseDoubleClickEvent(self, ev):
+        IndividualERPDialog(self)
+
+
+# Individual Event-Related Potential dialog ---------------------------------
+class IndividualERPDialog(QtGui.QDialog):
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:2122]: this shouldn't be here, it has been moved out to its own module, 'event_related_potential.py'

File: ecogvis/functions/misc_dialogs.py [hunk 10]
```diff
@@ -1541,9 +1598,1119 @@ def paint(self):
         self.GLViewWidget.renderText(self.X, self.Y, self.Z, self.text)
 
 
-# Gray line for visual separation of buttons -----------------------------------
+# Creates Event-Related Potential dialog ------------------------------------
+class ERPDialog(QMainWindow):
```
Comment (luiztauffer) [inline ecogvis/functions/misc_dialogs.py:1602]: this shouldn't be here, this class has been moved to its own module 'event_related_potential.py'

**Comment by luiztauffer** [comment]
I wouldn't merge it now. I ran it and has some weird bugs in some elements

File: ecogvis/functions/misc_dialogs.py (hunk 8 above)
Comment (luiztauffer) [reply ecogvis/functions/misc_dialogs.py:2691]: OK, I see that they have made changes to these functions. I rather we keep their changes and discard mine (which are just moving these classes to their own module)

**Comment by luiztauffer** [comment]
Maybe everyone should check if this version is working as expected for their current workflows? Before merging to the master
@bendichter 
@jgmakin 
@jrliu95 
@emilyps14 
@Armin12

File: ecogvis/functions/misc_dialogs.py (hunk 10 above)
Comment (luiztauffer) [reply ecogvis/functions/misc_dialogs.py:1602]: ignore that, we can move it later, the most important is to keep their changes

File: ecogvis/functions/misc_dialogs.py (hunk 2 above)
Comment (jrliu95) [reply ecogvis/functions/misc_dialogs.py:198]: yeah, this looks like a PyCharm type of automatic reformatting to fit PEP8 style with 79 chars per line. they may look strange but it shouldn't have resulted in any errors right?

**Comment by emilyps14** [comment]
This seems trivial but it's my workflow to run ecogVIS from inside a python console in PyCharm as:
`from ecogvis.ecogvis import main; main(<path to file>)`

This now breaks, because of the `ArgumentParser` usage within `main()`. My plan is to move the offending code to the `if __name__ == '__main__':` clause, since that will only run when ecogvis is called from the command line. 

Objections?

**Comment by bendichter** [comment]
It looks like some issues came up in the merge beyond weird formatting. I think this is going to few days to sort out. We'll be in touch when it's ready to test.

File: ecogvis/functions/misc_dialogs.py (hunk 2 above)
Comment (luiztauffer) [reply ecogvis/functions/misc_dialogs.py:198]: that's right, it shouldn't incur any errors

**Comment by luiztauffer** [comment]
I fixed some test errors and most are passing now. 
But there are specifically two that are still failing:
```
<40> np.testing.assert_array_almost_equal(speakerEventDS, speakerEventDS_expected)
<42> np.testing.assert_array_almost_equal(micEventDS, micEventDS_expected)
```

**Comment by bendichter** [comment]
@luiztauffer Thanks. Yes, I suspect there have been some changes to the event detection function that will break numerical equivalence. As long as the new results are reasonable we can just replace the array they are being compared to. I'd like to strictly adhere to the tests passing before merge so that we can continue to rely in CI to catch our mistakes. I'll take a look and see if I can repair the tests.

**Comment by luiztauffer** [comment]
@bendichter 
I fixed the event detection tests. The test data (a numpy array) didn’t make sense to me, so I’ve added some fake recording for mic and speaker (wav files), now the tests are passing just fine.

**Pull request merged**